Source/Core/Core/AchievementManager.cpp

```diff
@@ -914,6 +916,26 @@ void AchievementManager::Request(const rc_api_request_t* request,
       });
 }
 
+// Currently, when rc_client calls the memory peek method provided in its constructor (or in
+// rc_client_set_read_memory_function) it will do so on the thread that calls DoFrame, which is
+// currently the host thread, with one exception: an asynchronous callback in the load game process.
+// This is done to validate/invalidate each memory reference in the downloaded assets, mark assets
+// as unsupported, and notify the player upon startup that theere are unsupported assets and how
+// many. As such, all that call needs to do is return the number of bytes that can be read with this
+// call. As we do not want to probe memory in an asynchronous call, I provide a separate method for
```

Comment:
It could be made clearer why we don't want to do this. Example wording:

```suggestion
// call. As only the CPU and host threads are allowed to read from memory, I provide a separate method for
```